gui/system/forms.py

```diff
@@ -2705,6 +2717,19 @@ class Meta:
         model = models.Certificate
 
 
+def is_valid_privatekey(certificate, privatekey):
+    pkey_obj = crypto.load_privatekey(crypto.FILETYPE_PEM, privatekey)
+    cert_obj = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
+    ctx = SSL.Context(SSL.TLSv1_METHOD)
```

Comment:
I do not know if it would get restricted or not? I was asking you as to how you came to this conclusion.

Also, does this really work for CAs? since it is only in the CertificateImportForm